### PR TITLE
[Breaking] Fix DataDog crash when running without analytics url

### LINKED ISSUE #547: Server crashes on boot if DataDog url not provided
The bot crashes on boot if the DataDog url is not provided. A check should be added to just return the service collection if there is nothing to set up, just how it's handled if there's no compiler/decompiler address.

https://github.com/discord-csharp/MODiX/blob/bc80c0bfcdf4d175819ed7ae2fcfc11477807f0e/Modix/Extensions/ServiceCollectionExtensions.cs#L150-L170

Edit: Someone tag Distilld.

Also, here's my temp work-around:
```cs
public static IServiceCollection AddStatsD(this IServiceCollection services, IHostingEnvironment environment)
{
	services.AddSingleton<IDogStatsd, DebugDogStatsd>();
	return services;

	/*
	var cfg = new StatsdConfig { Prefix = "modix" };

	if (!environment.IsProduction() && string.IsNullOrWhiteSpace(cfg.StatsdServerName))
	{
		services.AddSingleton<IDogStatsd, DebugDogStatsd>();
		return services;
	}

	DogStatsd.Configure(cfg);
	services.AddSingleton(cfg);
	services.AddSingleton<IDogStatsd>(provider =>
	{
		var config = provider.GetRequiredService<StatsdConfig>();
		var service = new DogStatsdService();
		service.Configure(config);
		return service;
	});
	return services;
	*/
}
```

### PR DESCRIPTION
**Pull request by Foxtrek64**
Fixes #547 

Currently, when running in production, if the `MODIX_DD_AGENT_HOST` environment variable is not set, the bot will crash.

This adds a switch env var, `MODIX_ENABLESTATSD`, which defaults to false if not provided.

@Cisien You will need to update the container to have this env var and `true` as the value when this change is merged.

@jrusbatch Here's your requested ping.

### CODE REVIEW

File: Modix/Extensions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -150,8 +150,10 @@ public static IServiceCollection AddModix(this IServiceCollection services)
         public static IServiceCollection AddStatsD(this IServiceCollection services, IHostingEnvironment environment)
         {
             var cfg = new StatsdConfig { Prefix = "modix" };
-
-            if (!environment.IsProduction() && string.IsNullOrWhiteSpace(cfg.StatsdServerName))
+            bool.TryParse(Environment.GetEnvironmentVariable("MODIX_ENABLESTATSD"), out var enableStatsd);
```
Comment (Cisien) [inline Modix/Extensions/ServiceCollectionExtensions.cs]: Consume the setting through IConfiguration with configBuilder.AddEnvironmentVariables("MODIX") if it's not already configured.

Comment (Foxtrek64) [reply Modix/Extensions/ServiceCollectionExtensions.cs]: Done

**Comment by Foxtrek64** [comment]
So for some reason it is *very* insistent on adding my change to the PromotionsService. I've tried adding it to gitignore, telling git that I want my own personal version, telling it to stop tracking it, and yet it persists.

If there's a way to ignore that file in the pull or to tell git to not submit that as a changed file in the commit, let me know and I'll fix it.

**Comment by Scott-Caldwell** [comment]
I just undid the changes to PromotionsService in your branch through GitHub magic.

The build is failing, though, because:
```
Startup.cs(78,22): error CS7036: There is no argument given that corresponds to the required formal parameter 'configuration' of 'ServiceCollectionExtensions.AddStatsD(IServiceCollection, IHostingEnvironment, IConfiguration)' [/src/Modix/Modix.csproj]
```

**Comment by Foxtrek64** [comment]
> 
> 
> I just undid the changes to PromotionsService in your branch through GitHub magic.
> 
> The build is failing, though, because:
> 
> > Startup.cs(78,22): error CS7036: There is no argument given that corresponds to the required formal parameter 'configuration' of 'ServiceCollectionExtensions.AddStatsD(IServiceCollection, IHostingEnvironment, IConfiguration)' [/src/Modix/Modix.csproj]

Oh perfect, thank you.

That's my bad. I forgot to push my changes to startup.cs.

I'll push that change and any bugfixes I find in a few hours.

File: Modix/Extensions/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -147,11 +148,13 @@ public static IServiceCollection AddModix(this IServiceCollection services)
...
 
-        public static IServiceCollection AddStatsD(this IServiceCollection services, IHostingEnvironment environment)
+        public static IServiceCollection AddStatsD(this IServiceCollection services, IHostingEnvironment environment, IConfiguration configuration)
         {
             var cfg = new StatsdConfig { Prefix = "modix" };
 
-            if (!environment.IsProduction() && string.IsNullOrWhiteSpace(cfg.StatsdServerName))
+            var enableStatsd = configuration.GetValue<bool>(nameof(ModixConfig.EnableStatsd));
+            if (!enableStatsd ||
+                environment.IsDevelopment() && string.IsNullOrWhiteSpace(cfg.StatsdServerName))
```
Comment (Inzanit) [inline Modix/Extensions/ServiceCollectionExtensions.cs]: This should be switched back to `!environment.IsProduction()`

Comment (Foxtrek64) [reply Modix/Extensions/ServiceCollectionExtensions.cs]: This is done.

**Pull request merged**